sympy/core/power.py

```diff
@@ -1232,6 +1233,17 @@ def _eval_is_algebraic_expr(self, syms):
         else:
             return True
 
+    def _eval_rewrite_as_exp(self, base, expo):
+        from sympy import exp, log, I
+        if not base.is_zero:
+            if base.has(Symbol):
+                # delay evaluation if expo is non symbolic
+                # (as exp(x*log(5)) automatically reduces to x**5)
+                return exp(log(base)*expo, evaluate=expo.has(Symbol))
+            else:
+                u, v = log(base).as_real_imag()
```

Comment:
I would still do deep=False here, or just use `log(abs(base)) + arg(base)*I`. rewrite shouldn't do deep changes to the expression that are unrelated to the function it is rewriting to.

Reply:
Okay, will make the changes along with the remaining tests.